configure.ac

```diff
@@ -54,7 +65,7 @@ AC_ARG_WITH([mpi],
             [AS_IF([test -d $withval/lib64], [mpi_libdir="lib64"], [mpi_libdir="lib"])
              CFLAGS="-I$withval/include $CFLAGS"
              LDFLAGS="-L$withval/$mpi_libdir $LDFLAGS"],
-            [])
+            [AS_IF([test "$BYPASS_MPICC_CHECK" == "yes"], [], AC_MSG_ERROR([--with-mpi= argument missing]))])
```

Comment:
I would actually ditch the entire handling of --with-mpi, because it's unlikely to work properly.  Finding the right flags to use to build against an MPI is usually more complicated than just the -I and -L listed here (which is why there's an mpicc).  I'd remove this entire section, the `-lmpi` in the AM_LDFLAGS in tests/Makefile.am, and let customers set MPICC to something useful if tests are desired.  You'd still need a check here for tests enabled but MPICC not found, but that's pretty simple.